clients/rospy/src/rospy/impl/masterslave.py

```diff
@@ -437,20 +437,22 @@ def _connect_topic(self, topic, pub_uri):
         #    individual socket timeouts, but this could potentially
         #    affect user code.
         socket.setdefaulttimeout(60.)
-        tries = 0
-        max_num_tries = 3
         success = False
-        while not success:
-            tries += 1
+        interval = 0.5 # seconds
+        # try to get the topic information until succes or the ROS node is shutdown.
+        # -> on connections problems while ROS node start

```

Comment:
What does `->` mean in this case, return or exit or something else? I'd prefer an explicit comment, like "while ROS node is running try to connect, exit on connection problems" (not sure if that statement is accurate, but it is clearer than the current comment).